app/controllers/projects_controller.rb

```diff
@@ -0,0 +1,41 @@
+class ProjectsController < ApplicationController
+
+	MAX_VOTES = 3
+
+	def create
+		@hackday = Hackday.find_by_id(params[:project][:hackday_id].to_i)
+		@project = @hackday.projects.build(project_params)
+		if @project.save
+			flash.now[:success] = "Project created!"
+		else
+			flash.now[:danger] = "Project failed to create"
+		end
+		redirect_to @hackday
+	end
+
+	def update
+		@project = Project.find_by_id(params[:id].to_i) 
+		@hackday = @project.hackday
+
+		if @hackday.closed?
+				flash[:danger] = "Hackday has already closed"
+		else
+			current_votes = cookies[:"current_votes_#{@project.hackday_id}"].to_i
+			if (current_votes >= MAX_VOTES)
+				flash[:danger] = "No votes lefted"
+			else
+				flash[:success] = "Vote succeed"
+				cookies[:"current_votes_#{@project.hackday_id}"] = current_votes + 1
```

Comment:
you should probably do this with an if @project.save so that it only does the cookie thing if the save is successful.